pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -387,6 +389,16 @@ func (rd *restoreDataProcessor) processRestoreSpanEntry(
 		)
 	}
 
+	// If the system tenant is restoring a guest tenant span, we don't want to
+	// forward all the restored data to now, as there may be importing tables in
+	// that span, that depend on the difference in timestamps on restored existing
+	// vs importing keys to rollback.
+	if writeAtBatchTS && kr.fromSystemTenant &&
+		(bytes.HasPrefix(entry.Span.Key, keys.TenantPrefix) || bytes.HasPrefix(entry.Span.EndKey, keys.TenantPrefix)) {
+		log.Warningf(ctx, "restoring span %s at its original timestamps because it is a tenant span", entry.Span)
+		writeAtBatchTS = false
+	}
```

Comment:
This seems like it deserves an integration test.

Reply:
bah, I was worried you'd say that since that's a tedious test to write, but yeah, we should have it.

Reply:
Thanks for going the extra mile! That was a bit more involved than I expected (I figured we'd just check the timestamps), but great to test the whole import-reversal-post-restore case.